plugins/woocommerce/src/Internal/Utilities/WebhookUtil.php

```diff
@@ -133,4 +134,23 @@ private function get_webhook_ids_for_user( int $user_id ): array {
 			)
 		);
 	}
+
+	/**
+	 * Gets the count of webhooks that are configured to use the Legacy REST API to compose their payloads.
+	 *
+	 * @return int
+	 */
+	public function get_legacy_webhooks_count(): int {
+		global $wpdb;
+
+		$cache_key = WC_Cache_Helper::get_cache_prefix( 'webhooks' ) . 'legacy_count';
+		$count     = wp_cache_get( $cache_key, 'webhooks' );
+
+		if ( false === $count ) {
+			$count = absint( $wpdb->get_var( "SELECT count( webhook_id ) FROM {$wpdb->prefix}wc_webhooks WHERE `api_version` < 1;" ) );
+			wp_cache_add( $cache_key, $count, 'webhooks' );
```

Comment:
I don't see anywhere that this cached value gets invalidated, and it also doesn't have an expiration. This could cause an inaccurate count on sites that use a persistent object cache.

One way to address this would be to just make the value expire after an hour or something. But you could also run `wp_cache_delete` when a webhook is updated or deleted...

Reply:
There's something weird going on here. Cache-wise this `get_legacy_webhooks_count` is just mimicking the behavior of `WC_Webhook_Data_Store::get_webhook_count`, the one that get the count for a given webhook status. A code search doesn't give any result of code that invalidates the cache for the already existing webhook counts; yet the count is always correct, even immediately after editing an existing webhook in a site with a persistent cache enabled (for both the existing counts based on status, and the new legacy count I added). How is that even possible?

Reply:
By the way, if you cant to test yourself locally with a persistent cache, an easy option is to install https://wordpress.org/plugins/docket-cache/

Reply:
Mystery solved! There are calls to `WC_Cache_Helper::invalidate_cache_group` in the `create`,  `update` and `delete` methods of `WC_Webhook_Data_Store`, so we're good to go. I hadn't found these earlier because I was searching for "delete", not "invalidate". 🤦🏻‍♂️